adapters/handlers/grpc/v1/batch_parse_request.go

```diff
@@ -87,17 +87,24 @@ func BatchFromProto(req *pb.BatchObjectsRequest, authorizedGetClass func(string,
 
 		var vectors models.Vectors = nil
 		if len(obj.Vectors) > 0 {
-			parsedVectors := make(map[string][][]float32)
+			parsedVectors := make(map[string][]float32)
+			parsedMultiVectors := make(map[string][][]float32)
 			for _, vec := range obj.Vectors {
-				parsedVectors[vec.Name] = append(parsedVectors[vec.Name], byteops.Float32FromByteVector(vec.VectorBytes))
+				switch vec.Type {
+				case *pb.VectorType_VECTOR_TYPE_UNSPECIFIED.Enum(), *pb.VectorType_VECTOR_TYPE_FP32.Enum():
+					parsedVectors[vec.Name] = byteops.Float32FromByteVector(vec.VectorBytes)
+				case *pb.VectorType_VECTOR_TYPE_COLBERT_FP32.Enum():
+					parsedMultiVectors[vec.Name] = append(parsedMultiVectors[vec.Name], byteops.Float32FromByteVector(vec.VectorBytes))
+				default:
+					// do nothing
+				}
```

Comment:
can we reuse the extract_vector function from adapters/handlers/grpc/v1/parse_search_request.go?

Reply:
I have added a task for this to revisit this part of code next week